salt/modules/zypperpkg.py

```diff
@@ -1563,6 +1586,15 @@ def install(name=None,
     cmd_install = ['install', '--auto-agree-with-licenses']
 
     cmd_install.append(kwargs.get('resolve_capabilities') and '--capability' or '--name')
+    # Install / patching / upgrade with vendor change support is only in SLE 15+  opensuse Leap 15+
+    if not novendorchange:
+        __zypper__(root=root).refresh_zypper_flags()
+        if __zypper__(root=root).inst_avc:
+            cmd_install.append("--allow-vendor-change")
```

Comment:
BTW, this though wasn't the intention of the original design. This should happen instead:

```python
__zypper__(root=root).allow_vendor_change(novendorchange). ....
```

So all the logic of the vendor change-or-not should be in the Zypper's class above instead.

Reply:
Good point. We'll address this in a follow-up PR. Thx!